backend-rust/src/graphql_api/search_result.rs

```diff
@@ -0,0 +1,370 @@
+                    SELECT MAX(height) as max_height, MIN(height) as min_height
+                    FROM blocks
+                    WHERE
+                        height = $1
+                        OR starts_with(hash, $2)
+                "#,
+                lower_case_query.parse::<i64>().ok(),
+                lower_case_query,
+            )
+            .fetch_one(pool)
+            .await?;
+            connection.has_previous_page =
+                result.max_height.map_or(false, |db_max| db_max > page_max_height);
+            connection.has_next_page =
+                result.min_height.map_or(false, |db_min| db_min < page_min_height);
+        }
+        Ok(connection)
+    }
+
+    async fn transactions(
+        &self,
+        ctx: &Context<'_>,
+        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come after the specified cursor.")]
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
+        before: Option<String>,
+    ) -> ApiResult<connection::Connection<String, Transaction>> {
+        let transaction_hash_regex: Regex = Regex::new(r"^[a-fA-F0-9]{1,64}$")
+            .map_err(|_| ApiError::InternalError("Invalid regex".to_string()))?;
+        let pool = get_pool(ctx)?;
+        let config = get_config(ctx)?;
+        let query = ConnectionQuery::<DescendingI64>::new(
+            first,
+            after,
+            last,
+            before,
+            config.transaction_connection_limit,
+        )?;
+        let mut connection = connection::Connection::new(false, false);
+        if !transaction_hash_regex.is_match(&self.query) {
+            return Ok(connection);
+        }
+        let lower_case_query = self.query.to_lowercase();
+        let mut row_stream = sqlx::query_as!(
+            Transaction,
+            r#"SELECT * FROM (
+                SELECT
+                    index,
+                    block_height,
+                    hash,
+                    ccd_cost,
+                    energy_cost,
+                    sender_index,
+                    type as "tx_type: DbTransactionType",
+                    type_account as "type_account: AccountTransactionType",
+                    type_credential_deployment as "type_credential_deployment: CredentialDeploymentTransactionType",
+                    type_update as "type_update: UpdateTransactionType",
+                    success,
+                    events as "events: sqlx::types::Json<Vec<Event>>",
+                    reject as "reject: sqlx::types::Json<TransactionRejectReason>"
+                FROM transactions
+                WHERE
+                    starts_with(hash, $5)
+                    AND $2 < index
+                    AND index < $1
```

Comment:
Not sure if searching by `transaction_index` is something that makes sense (or is used by people). The old backend does not search by index so would keep that behavior.

Reply:
This is not entirely true. It does not provide the cursors, however it is indeed true that I highly doubt that this is a useable feature. To be consistent I would like to keep it as is.